scheds/rust/scx_flash/src/bpf/main.bpf.c

```diff
@@ -825,7 +825,6 @@ void BPF_STRUCT_OPS(flash_dispatch, s32 cpu, struct task_struct *prev)
 	    is_task_locked(prev)) {
 		task_unlock(prev);
 		task_refill_slice(prev);
-		return;
```

Comment:
Hm... if we just remove this `return` here, we would refill prev's time slice, but then call `scx_bpf_dsq_move_to_local()`, which can move the next task to the local DSQ (if there's any task in `SHARED_DSQ`). This can nullify refilling prev's time slice, as we would simply execute the next task upon returning from `ocps.dispatch()`.

In practice, this effectively ignores whether the task is holding a lock or not, which could potentially be better in some scenarios. Perhaps it might even make sense to remove the entire futex priority inversion detection logic altogether...

Can you check if you get the same performance by completely removing the whole `is_task_locked()` logic? I expect that you get the same performance, because your last change is pretty much doing this. Thanks!

Reply:
No problem thanks for the idea ! Let me test it real soon, I should look into what `scx_bpf_dsq_move_to_local()` actually performed deeper, I think I misunderstood some parts.

Removing futex detection to be better is in fact the opposite of my expectation because I do think making futex owner to finish faster will be a benefit , maybe the workload is heavy but can't create enough contention ? I'll check it for more detailed and get back to you ASAP. Thanks!

Reply:
After removing `is_task_locked()` part, the output shows similar performance.
```
$ pgbench -h localhost -p 5432 -U postgres -c 100 -j 16 -T 120 -M prepared --protocol extended testdb
Password: 
pgbench (16.6 (Ubuntu 16.6-0ubuntu0.24.04.1), server 15.10 (Debian 15.10-1.pgdg120+1))
starting vacuum...end.
transaction type: <builtin: TPC-B (sort of)>
scaling factor: 1000
query mode: extended
number of clients: 100
number of threads: 16
maximum number of tries: 1
duration: 120 s
number of transactions actually processed: 520080
number of failed transactions: 0 (0.000%)
latency average = 23.071 ms
initial connection time = 275.107 ms
tps = 4334.413322 (without initial connection time)
```

with the current implementation still perform worse ( I even execute more tests on it trying to avoid cache issue. )
```
$ pgbench -h localhost -p 5432 -U postgres -c 100 -j 16 -T 120 -M prepared --protocol extended testdb
Password: 
pgbench (16.6 (Ubuntu 16.6-0ubuntu0.24.04.1), server 15.10 (Debian 15.10-1.pgdg120+1))
starting vacuum...end.
transaction type: <builtin: TPC-B (sort of)>
scaling factor: 1000
query mode: extended
number of clients: 100
number of threads: 16
maximum number of tries: 1
duration: 120 s
number of transactions actually processed: 388329
number of failed transactions: 0 (0.000%)
latency average = 30.847 ms
initial connection time = 248.118 ms
tps = 3241.852846 (without initial connection time)
```

Also I found something interesting that in `scx_flash`, actually only `fexit_futex_wake` and `fexit__futex_wait` are triggered, others are almost silence, at least for this workload.

<img width="1460" alt="Screenshot 2024-12-24 at 12 44 21 AM" src="https://github.com/user-attachments/assets/46ee08ad-c679-4178-a4e0-b7578fa72a71" />


Maybe the deadline policy itself is already good enough for futex owner to finish their critical section in 1 or 2 period so the benefit of it isn't really sufficient in this scenario ?

Reply:
Probably for this particular workload the futex tracking is likely not very significant and might actually degrade performance.

Maybe we should introduce a command line option to enable/disable it, so that people can decided if they want the futex tracking or not.